sinks/cloudwatch/cloudwatch_test.go

```diff
@@ -249,3 +250,48 @@ func TestFlushRemoteTimeout(t *testing.T) {
 
 	<-done
 }
+
+func TestDroppedMetricsOnFail(t *testing.T) {
+	// Server handler should return response after the flush timeout expires
+	customTimeout := time.Millisecond * 500
+	serverDelay := customTimeout + time.Second
+
+	// Listen for PutMetricData
+	reqBodyCh := make(chan []byte)
+	server := NewTestServer(t, serverDelay, reqBodyCh)
+	defer server.Close()
+
+	// input.1.json contains three timeseries samples in InterMetrics format
+	jsInput, err := ioutil.ReadFile("testdata/input.1.json")
+	assert.NoError(t, err)
+	var metrics []samplers.InterMetric
+	assert.NoError(t, json.Unmarshal(jsInput, &metrics))
+
+	// Initialize Sink with a small timeout to simulate server issue
+	sink := NewCloudwatchMetricSink(
+		"cloudwatch", server.URL, "test", "us-east-1000", "cloudwatch_standard_unit",
+		customTimeout, true, []string{}, logrus.NewEntry(logrus.New()),
+	)
+	sink.Start(nil)
+
+	// Simulate the request to the server.
+	done := make(chan bool)
+	go func(reqBodyCh chan []byte, done chan bool) {
+		<-reqBodyCh
+		done <- true
+	}(reqBodyCh, done)
+
+	// Flush the sink
+	flushResult, err := sink.Flush(context.Background(), metrics)
+	assert.Error(t, err)
+	assert.Equal(t,
```

Comment:
Instead of writing this as a separate test case, can we assert the returned value of `flushResult` is correct in every other test case in this file?

Reply:
I think `TestFlushRemoteTimeout` and my test case `TestDroppedMetricsOnFail` can be consolidated in that case. I'm relying on the timeout to simulate the server as unresponsive.